bin/opd8n/src/cmd/from_l1.rs

```diff
@@ -57,29 +54,24 @@ impl FromL1 {
             self.end_block > self.start_block,
             "End block must come after the start block"
         );
-        let outputs = if self.derive {
-            self.derive().await?
-        } else {
-            self.outputs.clone()
-        };
-        ensure!(
-            !outputs.is_empty(),
-            "Must provide at least one L2 output root"
-        );
         trace!(target: "from-l1", "Producing derivation fixture for L1 block range [{}, {}]", self.start_block, self.end_block);
 
         // Construct a sequential list of block numbers from [start_block, end_block].
         let blocks = (self.start_block..=self.end_block).collect::<Vec<_>>();
 
         // Construct the providers
         let l1_rpc_url =
-            Url::parse(&self.rpc_url).map_err(|e| eyre!("Invalid L1 RPC URL: {}", e))?;
+            Url::parse(&self.l1_rpc_url).map_err(|e| eyre!("Invalid L1 RPC URL: {}", e))?;
         let mut l1_provider = AlloyChainProvider::new_http(l1_rpc_url);
-        let l1_chain_id = l1_provider.chain_id().await.map_err(|e| eyre!(e))?;
-        info!(target: "from-l1", "Using L1 Chain ID: {}", l1_chain_id);
+        let l2_rpc_url =
+            Url::parse(&self.l2_rpc_url).map_err(|e| eyre!("Invalid L1 RPC URL: {}", e))?;
+        let mut l2_provider =
+            AlloyL2ChainProvider::new_http(l2_rpc_url, Arc::new(Default::default()));
```

Comment:
Consider simplifying to the following
```suggestion
        let mut l1_provider = AlloyChainProvider::new_http(&self.l1_rpc_url.parse());

        let mut l2_provider =
            AlloyL2ChainProvider::new_http(&self.l2_rpc_url.parse()?, Arc::new(Default::default()));
```